Formula/m4.rb

```diff
@@ -32,6 +32,23 @@ class M4 < Formula
     end
   end
 
+  if OS::Linux::Glibc.system_version >= "2.28" ||
+     (Formula["glibc"].any_version_installed? && Formula["glibc"].version >= "2.28")
+    # Glibc no longer provides several required symbols since version 2.28.
+    # - https://sourceware.org/legacy-ml/libc-announce/2018/msg00000.html
+    # As a result gnulib, bundled into m4, and the dependent of those
+    # removed symbols would no long compile until an upstream fix was applied.
+    # - https://github.com/coreutils/gnulib/commit/4af4a4a71827c0bc5e0ec67af23edef4f15cee8e#diff-5bcce8ce55246264586c4aed2a45ff89
+    # Version 1.4.18 of m4, however, includes a bundled copy of gnulib that
+    # does not contain the upstream patch. Patch the copy of gnulib bundled
+    # with the m4 distribution.
+    # - https://lists.gnu.org/archive/html/bug-m4/2018-08/msg00000.html
+    patch do
+      url "https://raw.githubusercontent.com/archlinux/svntogit-packages/19e203625ecdf223400d523f3f8344f6ce96e0c2/trunk/m4-1.4.18-glibc-change-work-around.patch"
+      sha256 "fc9b61654a3ba1a8d6cd78ce087e7c96366c290bc8d2c299f09828d793b853c8"
+    end
+  end
```

Comment:
```suggestion
  on_linux do
    if OS::Linux::Glibc.system_version >= "2.28" ||
       (Formula["glibc"].any_version_installed? && Formula["glibc"].version >= "2.28")
      # Glibc no longer provides several required symbols since version 2.28.
      # - https://sourceware.org/legacy-ml/libc-announce/2018/msg00000.html
      # As a result gnulib, bundled into m4, and the dependent of those
      # removed symbols would no long compile until an upstream fix was applied.
      # - https://github.com/coreutils/gnulib/commit/4af4a4a71827c0bc5e0ec67af23edef4f15cee8e#diff-5bcce8ce55246264586c4aed2a45ff89
      # Version 1.4.18 of m4, however, includes a bundled copy of gnulib that
      # does not contain the upstream patch. Patch the copy of gnulib bundled
      # with the m4 distribution.
      # - https://lists.gnu.org/archive/html/bug-m4/2018-08/msg00000.html
      patch do
        url "https://raw.githubusercontent.com/archlinux/svntogit-packages/19e203625ecdf223400d523f3f8344f6ce96e0c2/trunk/m4-1.4.18-glibc-change-work-around.patch"
        sha256 "fc9b61654a3ba1a8d6cd78ce087e7c96366c290bc8d2c299f09828d793b853c8"
      end
    end
  end
```
Might need this OS check; not sure.

Reply:
Yea, I think you need it:
```
❯ gh pr checkout 66763
❯ brew info m4
Error: No available formula or cask with the name "m4".
```

Reply:
Thank you @carlocab . I've accepted the suggestion.